File: 2025/Company/Company.Web/Modules/ServerTypes/Company/ProvinceColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProvinceRow } from "./ProvinceRow";

export interface ProvinceColumns {
    ProvinceId: Column<ProvinceRow>;
    ProvinceCode: Column<ProvinceRow>;
    CountryId: Column<ProvinceRow>;
    ProvinceName: Column<ProvinceRow>;
}

export class ProvinceColumns extends ColumnsBase<ProvinceRow> {
    static readonly columnsKey = 'Company.Province';
    static readonly Fields = fieldsProxy<ProvinceColumns>();
}

File: 2025/Company/Company.Web/Modules/Company/Province/ProvinceColumns.cs
namespace Company.Company.Columns;

[ColumnsScript("Company.Province")]
[BasedOnRow(typeof(ProvinceRow), CheckNames = true)]
public class ProvinceColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ProvinceId { get; set; }
    [EditLink]
    public string ProvinceCode { get; set; }
    [Width(130)]
    [AsyncLookupEditor(typeof(CountryLookup)), QuickFilter]
    public int CountryId { get; set; }
    //public string CountryName { get; set; }
    public string ProvinceName { get; set; }
}